tests/integrated/test_suite

```diff
@@ -58,6 +63,98 @@ try:
 except RuntimeError:
     raise RuntimeError("The all-tests flag needs to be set by passing --all rather then --set-bool all_tests=True")
 
+##################################################################
+# Parallel stuff
+
+# Check what we are supposed to be doing:
+if 'MAKEFLAGS' in os.environ:
+    mf=os.environ['MAKEFLAGS']
+    for token in mf:
+        if token == "n":
+            # Print recipies:
+            # We ignore this for now ...
+            sys.exit(0)
+        elif token == "q":
+            # Question mode:
+            # We are never up-to-date
+            sys.exit(1)
+        elif token in "ksBd":
+            # k - keep going
+            # s - silent
+            # B - always build (we do)
+            # d - debug
+            pass
+        elif token == " ":
+            break
+        elif token == "-":
+            # This is not from make
+            break
+        else:
+            # Not implemented
+            print("mode '%s' not implemented"%token)
+            #sys.exit(42)
+else:
+    mf = ""
+
+num_jobs=1
+if args.jobs:
+    num_jobs = args.jobs[0]
+js_read=None
+if "--jobserver-auth=" in mf or "--jobserver-fds=" in mf:
+    # running in parallel
+    try:
+        offset=mf.index("--jobserver-auth=")
+    except ValueError:
+        offset=mf.index("--jobserver-fds=")
+    file_desc=[int(i) for i in re.findall("\d+",mf[offset:])[0:2]]
+    js_read = file_desc[0]
+    js_write = file_desc[1]
+
+class Test(threading.Thread):
+    def __init__(self, name, cmd):
+        threading.Thread.__init__(self)
+        self.name = name
+        self.cmd = cmd
+        self.l = threading.local()
+        self.cost = self._cost()
+    def run(self):
+        # Check requirements
+        self.l.req_met, self.l.req_expr = requirements.check(self.name+"/runtest")
+        self.l.start_time = time.time()
+        if not self.l.req_met:
+            print(output % self.name +
+                  "S - {0} => False".format(self.l.req_expr))
+            sys.stdout.flush()
+            self.status = 0
+        else:
+            # Run test, piping stdout so it is not sent to console
+            self.status, self.output = shell("cd "+self.name+";"+self.cmd, pipe=True)
+            print(output % self.name, end='')
+            if self.status != 0:
+                # ❌ Failed
+                print(u"\u274C", end='')  # No newline
+            else:
+                # ✓ Passed
+                print(u"\u2713", end='')  # No newline
+            print(" %7.3f s" % (time.time() - self.l.start_time))
+            sys.stdout.flush()  # Make sure '✓' or '❌' is flushed
+    def _cost(self):
```

Comment:
Could make this a `@property` and remove the `self.cost = _cost()` from above?

Reply:
I think `@property` wouldn't cache the cost?
Also, a setter would be nice. So after reading about `@property`, I don't think it makes sense here.